lib/meadow/pipeline/actions/generate_file_set_digests.ex

```diff
@@ -25,15 +25,21 @@ defmodule Meadow.Pipeline.Actions.GenerateFileSetDigests do
     try do
       Logger.info("Generating digests for #{file_set.id}")
 
-      hashes = generate_hashes(file_set.metadata.location)
-      Logger.info("Hash for #{file_set.id}: #{hashes.sha256}")
-
-      {:ok, _} =
-        file_set
-        |> FileSets.update_file_set(%{metadata: %{digests: hashes}})
-
-      ActionStates.set_state!(file_set, __MODULE__, "ok")
-      :ok
+      case generate_hashes(file_set.metadata.location) do
+        {:ok,
+         %{"sha256" => <<sha256::binary-size(64)>>, "sha1" => <<_sha1::binary-size(40)>>} = hashes} ->
+          Logger.info("Hash for #{file_set.id}: #{sha256}")
+          {:ok, _} = FileSets.update_file_set(file_set, %{metadata: %{digests: hashes}})
+          ActionStates.set_state!(file_set, __MODULE__, "ok")
+          :ok
+
+        {:ok, unexpected_result} ->
+          ActionStates.set_state!(file_set, __MODULE__, "error", unexpected_result)
+          {:error, unexpected_result}
+
+        {:error, error} ->
```

Comment:
I feel like we need to handle the case where the return is `_` (anything else) and store an error for this action

Reply:
I think that's handled between the `{:ok, unexpected_result}` case and the `{:error, error}` case, right?

Reply:
We might. It looks like `Lambda.invoke` could return `other -> other` and I don't think we have any idea what that is. It might not be covered with our cases, right?

Reply:
I guess there is a tiny chance that we could get back something that doesn't match either `{:ok, ...}` or `{:error, ...}`. Maybe a case for

```
very_unexpected_result -> 
    ActionStates.set_state!(file_set, __MODULE__, "error", very_unexpected_result)
    {:error, very_unexpected_result}
```

??

Reply:
But wait... `Meadow.Utils.Lambda` is already making sure that everything is coming back as `{:ok, something}` or `{:error, something}` by contract. We don't need the third case.

Reply:
where exactly is this happening @mbklein ?